components/brave_wallet_ui/common/hardware/ledgerjs/eth_ledger_bridge_keyring.ts

```diff
@@ -1,152 +1,151 @@
-      const signed = await eth.signTransaction(path, rawTxHex)
-      return { success: true, payload: signed }
-    } catch (e) {
-      return { success: false, error: e.message, code: e.statusCode || e.id || e.name }
+    const responsePayload = data.payload as EthSignPersonalMessageResponsePayload
+    const signature = this.createMessageSignature(responsePayload)
+    if (!signature) {
+      return { success: false, error: getLocale('braveWalletLedgerValidationError') }
     }
+
+    return { success: true, payload: signature }
   }
 
   signEip712Message = async (path: string, domainSeparatorHex: string, hashStructMessageHex: string): Promise<SignHardwareOperationResult> => {
-    try {
-      const unlocked = await this.unlock()
-      if (!unlocked.success || !this.app) {
-        return unlocked
-      }
-      const eth: Eth = this.app
-      const data = await eth.signEIP712HashedMessage(path, domainSeparatorHex, hashStructMessageHex)
-      const signature = this.createMessageSignature(data)
-      return { success: true, payload: signature }
-    } catch (e) {
-      return { success: false, error: e.message, code: e.statusCode || e.id || e.name }
+    const result = await this.unlock()
+    if (!result.success) {
+      return result
     }
-  }
-
-  signPersonalMessage = async (path: string, message: string): Promise<SignHardwareOperationResult> => {
-    try {
-      const unlocked = await this.unlock()
-      if (!unlocked.success || !this.app) {
-        return unlocked
-      }
-      const eth: Eth = this.app
-      const messageHex = Buffer.from(message).toString('hex')
-      const data = await eth.signPersonalMessage(path, messageHex)
-      const signature = this.createMessageSignature(data)
-      if (!signature) {
-        return { success: false, error: getLocale('braveWalletLedgerValidationError') }
-      }
-      return { success: true, payload: signature }
-    } catch (e) {
-      return { success: false, error: e.message, code: e.statusCode || e.id || e.name }
+    const data = await this.sendCommand<EthSignEip712MessageResponse>({
+      command: LedgerCommand.SignEip712Message,
+      id: LedgerCommand.SignEip712Message,
+      path: path,
+      origin: window.origin,
+      domainSeparatorHex: domainSeparatorHex,
+      hashStructMessageHex: hashStructMessageHex
+    })
+    if (data === LedgerBridgeErrorCodes.BridgeNotReady ||
+        data === LedgerBridgeErrorCodes.CommandInProgress) {
+      return this.createErrorFromCode(data)
+    }
+    if (!data.payload.success) {
+      const ledgerError = data.payload as LedgerError
+      return { success: false, error: ledgerError.message, code: ledgerError.statusCode }
     }
-  }
-
-  cancelOperation = async () => {
-    this.app?.transport.close()
-  }
 
-  private onDisconnected = (e: any) => {
-    if (e.name !== 'DisconnectedDevice') {
-      return
+    const responsePayload = data.payload as EthSignEip712MessageResponsePayload
```

Comment:
same nit here for type casting

Reply:
https://github.com/brave/brave-core/pull/14373#discussion_r942894857